packages/storybook/src/documentation/pages/demos/pagination/serverPagination.demo.tsx

```diff
@@ -0,0 +1,74 @@
+import { Columns, FeatureMode, PageChangedParams, RowsProp, XGrid } from '@material-ui/x-grid';
+import { useCallback, useState } from 'react';
+import * as React from 'react';
+import {
+  randomCreatedDate,
+  randomEmail,
+  randomId,
+  randomInt,
+  randomTraderName,
+  randomUpdatedDate,
+} from '@material-ui/x-grid-data-generator';
+
+const newRow = () => ({
+  id: randomId(),
+  name: randomTraderName(),
+  email: randomEmail(),
+  age: randomInt(10, 100),
+  dateCreated: randomCreatedDate(),
+  lastLogin: randomUpdatedDate(),
+});
+
+const columns: Columns = [
+  { field: 'id', hide: true },
+  { field: 'name', type: 'string' },
+  { field: 'email', type: 'string' },
+  { field: 'age', type: 'number' },
+  { field: 'dateCreated', type: 'date', width: 180 },
+  { field: 'lastLogin', type: 'dateTime', width: 180 },
+];
+
+function loadServerRows(params: PageChangedParams): Promise<any> {
+  return new Promise<any>((resolve) => {
+    const rows: any[] = [];
+    while (rows.length < params.pageSize) {
+      rows.push(newRow());
+    }
+
+    setTimeout(() => {
+      resolve(rows);
+    }, 800);
+  });
+}
+
+export default function ServerPaginationDemo() {
+  const [rows, setRows] = useState<RowsProp>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
+
+  const onPageChanged = useCallback(
```

Comment:
Why do we need useCallback?

Reply:
otherwise you change the prop at every render

Reply:
This isn't an issue by itself. Why is it important?

For context, so far, no components requires
 the props to be memoized to be behave with acceptable performance. From experience, developers struggles to guarantee it in the first place.

I see two possible outcomes:

1. useCallback has no measurable performance downsides, we should drop it, we should never add performance logic unless we identify a **bottleneck**. No early optimizations.
2. There is a performance implication, we should look into removing the memoization requirement. We have found the `useEventCallback` custom hook to solve this problem in the past.

Reply:
That is what useCallback does, yes. I think Olivier was more asking why we need it here. In the end it's passed to `XGrid` that receives a new `options` object on every render. Maybe XGrid doesn't use the default arePropsEqual implementation of `React.memo`? Or is that specific option passed to another component where a re-render is expensive?

Reply:
Because if you send a new ref all the time, then XGrid is going to treat as a new handler as it's a new function and thus rebind it to the event.

Reply:
So we can remove useCallback in such a case?

Reply:
No otherwise you will rebind the event all the time.

Reply:
Why is this an issue? Does it mean we should use `useEventCallback`?

Reply:
Well it's like calling addEventListener every time you render. It's a bad practice. 
So at every render, it will drop the old handler and replace it with the new one. The demo works without the useCallback so I can do the change if you want. 

Why would we need to use `useEventCallback` over the native `useCallback`. What's the issue in using `useCallback`?

Reply:
Yeah, agree, we have seen issues reported by users on the main repository around the addEventListener repeating call you are mentioning. 

> What's the issue in using useCallback?

My fear is that developers won't successfully manage to enforce this constraint at scale (meaning as they refactor the code). Now, It will still be interesting to see how that turns out. I think that we can keep useCallback and optimize for learning, if we see people raising bugs because they didn't implement memorization correctly (my assumption and our experience is that they will), then, at this point we can reconsider.